packages/devtools_app/lib/src/shared/console/widgets/evaluate.dart

```diff
@@ -315,7 +308,13 @@ class ExpressionEvalFieldState extends State<ExpressionEvalField>
     try {
       // Response is either a ErrorRef, InstanceRef, or Sentinel.
       final isolateRef = serviceManager.isolateManager.selectedIsolate.value;
-      final response = await evalService.evalAtCurrentFrame(expressionText);
+      final Response response;
+      if (serviceManager.isMainIsolatePaused) {
+        response = await evalService.evalAtCurrentFrame(expressionText);
+      } else {
+        response =
+            await evalService.evalInRunningApp(isolateRef!, expressionText);
```

Comment:
should we be null asserting isolateRef? asking because _emitRefToConsole allows a nullable isolate below. Maybe we should add a null check if we cannot call the eval on a null isolate